cpp/src/replace/nulls.cu

```diff
@@ -352,6 +356,75 @@ std::unique_ptr<cudf::column> replace_nulls_scalar_kernel_forwarder::operator()<
   return cudf::dictionary::detail::replace_nulls(dict_input, replacement, stream, mr);
 }
 
+struct replace_nulls_replace_policy_functor {
+  __device__ thrust::tuple<cudf::size_type, bool> operator()(
+    thrust::tuple<cudf::size_type, bool> const& lhs,
+    thrust::tuple<cudf::size_type, bool> const& rhs)
+  {
+    return thrust::get<1>(rhs) ? thrust::make_tuple(thrust::get<0>(rhs), true)
+                               : thrust::make_tuple(thrust::get<0>(lhs), true);
+  }
+};
+
+/**
+ * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+ *        `replace_nulls` with the appropriate data types.
+ */
+struct replace_nulls_replace_policy_kernel_forwarder {
+  template <typename col_type,
+            typename std::enable_if_t<cudf::is_fixed_width<col_type>()>* = nullptr>
+  std::unique_ptr<cudf::column> operator()(cudf::column_view const& input,
+                                           cudf::replace_policy const& replace_policy,
+                                           rmm::cuda_stream_view stream,
+                                           rmm::mr::device_memory_resource* mr)
+  {
+    using Type     = cudf::device_storage_type_t<col_type>;
+    auto device_in = cudf::column_device_view::create(input);
+    auto index     = thrust::make_counting_iterator<cudf::size_type>(0);
+    auto valid_it  = cudf::detail::make_validity_iterator(*device_in);
+    auto in_begin  = thrust::make_zip_iterator(thrust::make_tuple(index, valid_it));
+
+    rmm::device_vector<cudf::size_type> gather_map(input.size());
```

Comment:
```suggestion
    rmm::device_uvector<cudf::size_type> gather_map(input.size());
```

Reply:
I think you want to initialize this gather_map with some sort of sentinel value that is out of bounds so elements in the output that should be null get nullified in the gather.